Documentation/teaching/labs/kernel_modules.rst

```diff
@@ -994,3 +994,148 @@ displayed processes differ. This is because a module is being loaded
 from the executable :file:`/sbin/insmod` when the module is loaded and
 when the module is unloaded a process is created from the executable
 :file:`/sbin/rmmod`.
+
+.. Extra
+   -----
+
+1. KDB
+------
+
+Go to the **9-kdb** directory. Activate KDB the over serial port and enter KDB
+mode using :command:`SysRq`. Connect to the pseudo-terminal linked to virtiocon0
+using :command:`minicom`, configure KDB to use the hvc0 serial port:
+
+.. code-block:: bash
+
+    echo hvc0 > /sys/module/kgdboc/parameters/kgdboc
+
+and enable it using SysRq (:command:`Ctrl + O g`).
+Review the current system status (:command:`help` to see the available KDB
+commands). Continue the kernel execution using the :command:`go` command.
+
+Load the :file:`hello_kdb` module.
+The module will simulate a bug when writing to the :file:`/proc/hello_kdb_bug`
+file. To simulate a bug, use the below command:
+
+.. code-block:: bash
+
+    echo 1 > /proc/hello_kdb_bug
+
+After running the above command, at every oops/panic the kernel stops the
+execution and enters debug mode.
+
+Analyze the stacktrace and determine the code that generated the bug.
+How can we find out from KDB the address where the module was loaded?
+
+In parallel, use GDB in a new window to view the code based on KDB information.
+
+.. hint::
+    Load the symbol file. Use :command:`info line`.
+
+When writing to :file:`/proc/hello_kdb_break`, the module will increment the
+:c:data:`kdb_write_address` variable. Enter KDB and set a breakpoint for each
+write access of the :c:data:`kdb_write_address` variable.
+Return to kernel to trigger a write using:
+
+.. code-block:: bash
+
+    echo 1 > /proc/hello_kdb_break
+
+2. PS Module
+------------
+
+Update the created kernel module at :ref:`proc-info` in order to display
+information about all the processes from the system, when inserting the kernel
+module, not just about the current process. Afterwards, compare the obtained
+result with the output of the :command:`ps` command.
+
+.. hint::
+    * Processes in the system are structured in a circular list.
+
+    * :c:macro:`for_each _...` macros (such as :c:macro:`for_each_process`) are
+      useful when you want to navigate the items in a list.
+
+    * To understand how to use a feature or a macro, use |LXR|_ or Vim and
+      :command:`cscope` and search for usage scenarios.
+
+3. Memory Info
+--------------
+
+Create a kernel module that displays the virtual memory areas of the current
+process; for each memory area it will display the start address and the end
+address.
+
+.. hint::
+    * Start from an existing kernel module.
+
+    * Investigate the structures
+      `struct task_struct <https://elixir.bootlin.com/linux/v4.15.7/source/include/linux/sched.h#L520>`_,
```

Comment:
Please don't use LXR links, use :c:type:`struct task_struct` so we can integrate with kernel docs.